pkg/models/notification/notification.go

```diff
@@ -0,0 +1,383 @@
+		return o.ksClient.NotificationV2().SlackReceivers().Create(context.Background(), obj.(*v2.SlackReceiver), v1.CreateOptions{})
+	case v2.ResourcesPluralWebhookConfig:
+		return o.ksClient.NotificationV2().WebhookConfigs().Create(context.Background(), obj.(*v2.WebhookConfig), v1.CreateOptions{})
+	case v2.ResourcesPluralWebhookReceiver:
+		return o.ksClient.NotificationV2().WebhookReceivers().Create(context.Background(), obj.(*v2.WebhookReceiver), v1.CreateOptions{})
+	case v2.ResourcesPluralWechatConfig:
+		return o.ksClient.NotificationV2().WechatConfigs().Create(context.Background(), obj.(*v2.WechatConfig), v1.CreateOptions{})
+	case v2.ResourcesPluralWechatReceiver:
+		return o.ksClient.NotificationV2().WechatReceivers().Create(context.Background(), obj.(*v2.WechatReceiver), v1.CreateOptions{})
+	default:
+		return nil, errors.NewInternalError(nil)
+	}
+}
+
+// Delete a object, if the user is nil, it only can delete the global object.
+// If the user is not nil, it only can delete the tenant object which the tenant
+// specified in the labels of the object equal to this user.
+func (o *operator) Delete(user, resource, name string) error {
+
+	if obj, err := o.Get(user, resource, name); err != nil {
+		return err
+	} else {
+		if err := authorizer(user, obj); err != nil {
+			return err
+		}
+	}
+
+	switch resource {
+	case v2.ResourcesPluralDingTalkConfig:
+		return o.ksClient.NotificationV2().DingTalkConfigs().Delete(context.Background(), name, v1.DeleteOptions{})
+	case v2.ResourcesPluralDingTalkReceiver:
+		return o.ksClient.NotificationV2().DingTalkReceivers().Delete(context.Background(), name, v1.DeleteOptions{})
+	case v2.ResourcesPluralEmailConfig:
+		return o.ksClient.NotificationV2().EmailConfigs().Delete(context.Background(), name, v1.DeleteOptions{})
+	case v2.ResourcesPluralEmailReceiver:
+		return o.ksClient.NotificationV2().EmailReceivers().Delete(context.Background(), name, v1.DeleteOptions{})
+	case v2.ResourcesPluralSlackConfig:
+		return o.ksClient.NotificationV2().SlackConfigs().Delete(context.Background(), name, v1.DeleteOptions{})
+	case v2.ResourcesPluralSlackReceiver:
+		return o.ksClient.NotificationV2().SlackReceivers().Delete(context.Background(), name, v1.DeleteOptions{})
+	case v2.ResourcesPluralWebhookConfig:
+		return o.ksClient.NotificationV2().WebhookConfigs().Delete(context.Background(), name, v1.DeleteOptions{})
+	case v2.ResourcesPluralWebhookReceiver:
+		return o.ksClient.NotificationV2().WebhookReceivers().Delete(context.Background(), name, v1.DeleteOptions{})
+	case v2.ResourcesPluralWechatConfig:
+		return o.ksClient.NotificationV2().WechatConfigs().Delete(context.Background(), name, v1.DeleteOptions{})
+	case v2.ResourcesPluralWechatReceiver:
+		return o.ksClient.NotificationV2().WechatReceivers().Delete(context.Background(), name, v1.DeleteOptions{})
+	default:
+		return errors.NewInternalError(nil)
+	}
+}
+
+// Update a object, if the user is nil, it only can update the global object.
```

Comment:
// Update a object, if the user is nil, it only can update the global object.
// If the user is not nil, it only can delete the tenant object which the tenant
// specified in the labels of the object equal to this user.
=>
// Update an object, only a global object will be updated if the user is nil.
// If the user is not nil, a tenant object whose tenant label matches the user will be updated